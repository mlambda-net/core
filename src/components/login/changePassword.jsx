import React from 'react';
import { Box, Button, FormControl, Paper, TextField } from '@mui/material';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/web-core/utils';
import { Title } from '@mlambda-net/web-core/common';

const language = new LocalizedStrings({
  en: {
    title: 'Change Password',
    newPassword: 'New Password',
    oldPassword: 'Old Password',
    email: 'Email',
    errorNewPassword: 'new password is required',
    errorOldPassword: 'old password is required',
    changeAction: 'submit',
  },
  es: {
    title: 'Cambiar contraseña',
    newPassword: 'Contraseña Nueva',
    oldPassword: 'Contraseña Actual',
    email: 'Email',
    errorNewPassword: 'la contraseña nueva es requerida',
    errorOldPassword: 'la contraseña actual es requerida',
    changeAction: 'cambiar',
  },
});

const styles = (theme) => ({});

class ChangePassword extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      newPassword: '',
      oldPassword: '',
      validOldPassword: false,
      validNewPassword: false,
      errorNewPassword: '',
      errorOldPassword: '',
    };

    this.valid = this.props.validation;
    this.newPasswordChange = this.newPasswordHandle.bind(this);
    this.oldPasswordChange = this.oldPasswordHandle.bind(this);
    this.changeClick = this.changeHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);
    const { classes } = this.props;
    return (
      <Paper elevation={10} className={this.props.className}>
        <Box p={3}>
          <Box display="flex" justifyContent="center">
            <Title title={language.title} />
          </Box>
          <Box
            display="flex"
            flexDirection="column"
            justifyContent="center"
            m={1}
            p={1}>
            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="password"
                  label={language.oldPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validOldPassword}
                  onChange={this.oldPasswordChange}
                  helperText={this.state.errorOldPassword}
                />
              </FormControl>
            </Box>

            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="confirm_password"
                  label={language.newPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validNewPassword}
                  onChange={this.newPasswordChange}
                  helperText={this.state.errorNewPassword}
                />
              </FormControl>
            </Box>

            <Box p={1} m={1} display="flex" justifyContent="flex-end">
              <Button
                variant="outlined"
                color="primary"
                disabled={!this.isValid()}
                onClick={this.changeClick}>
                {language.changeAction}
              </Button>
            </Box>
          </Box>
        </Box>
      </Paper>
    );
  }

  newPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorNewPassword: language.errorNewPassword,
      });
    } else {
      this.setState({
        newPassword: password,
        validNewPassword: true,
        errorNewPassword: '',
      });
    }
  }

  oldPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorOldPassword: language.errorOldPassword,
      });
    } else {
      this.setState({
        oldPassword: password,
        validOldPassword: true,
        errorOldPassword: '',
      });
    }
  }

  changeHandle(event) {
    this.props.onChange({
      newPassword: this.state.newPassword,
      oldPassword: this.state.oldPassword,
    });

    event.preventDefault();
  }

  isValid() {
    return this.state.validNewPassword && this.state.validOldPassword;
  }
}

export default withUtils(styles)(ChangePassword);
