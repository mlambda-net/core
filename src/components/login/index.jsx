export { default as Register } from './register';
export { default as SignIn } from './signin';
export { default as ChangePassword } from './changePassword';
export { default as Recovery } from './recovery';
export { default as Profile } from './profile';
