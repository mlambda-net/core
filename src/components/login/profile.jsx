import React from 'react';
import LocalizedStrings from 'react-localization';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import { Box, Card, CardActions, CardContent, Typography } from '@mui/material';
import { Option } from '@mlambda-net/web-core/common';
import { withUtils } from '@mlambda-net/web-core/utils';

const language = new LocalizedStrings({
  en: {
    title: 'Profile',
  },
  es: {
    title: 'Perfil',
  },
});

const styles = (theme) => ({
  panel: {
    width: '100%',
    height: '100%',
  },

  title: {
    textAlign: 'center',
  },
  content: {
    height: 'calc(100% - 80px)',
  },
  actions: {
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'flex-end',
  },
});

class Profile extends React.Component {
  constructor(props) {
    super(props);
    this.valid = this.validation;
    this.state = { open: false };
  }

  render() {
    language.setLanguage(this.props.lang);
    const Content = this.props.content;
    const Actions = this.props.actions;
    const { classes } = this.props;

    return (
      <Option
        icon={AccountCircleIcon}
        width={this.props.width}
        height={this.props.height}
        style={this.props.style}>
        <Card className={classes.panel}>
          <CardContent className={classes.content}>
            <Typography
              className={classes.title}
              color="textSecondary"
              variant="h5"
              gutterBottom>
              {language.title}
            </Typography>
            <Content user={this.props.user} />
          </CardContent>

          <CardActions className={classes.actions}>
            <Box>
              <Actions />
            </Box>
          </CardActions>
        </Card>
      </Option>
    );
  }
}

export default withUtils(styles)(Profile);
