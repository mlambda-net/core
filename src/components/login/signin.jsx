import React from 'react';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/web-core/utils';
import {
  Box,
  Button,
  FormControl,
  Link,
  TextField,
  Typography,
} from '@mui/material';

const language = new LocalizedStrings({
  en: {
    password: 'Password',
    email: 'Email',
    invalidEmail: 'invalid email',
    invalidPassword: 'password is required',
    forgot: 'forgot password?',
    login: 'login',
  },
  es: {
    password: 'Contraseña',
    email: 'Correo',
    invalidEmail: 'correo invalido',
    invalidPassword: 'una contraseña es requerida',
    forgot: 'olvido la contraseña?',
    login: 'ingresar',
  },
});

const styles = (theme) => ({});

class SignIn extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      validEmail: false,
      validPassword: false,
      helpEmail: '',
      helpPassword: '',
    };
    this.valid = this.props.validation;
    this.emailChange = this.emailChangeHandle.bind(this);
    this.passwordChange = this.passwordChangeHandle.bind(this);
    this.loginClick = this.loginHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);

    return (
      <Box width="100%">
        <Box display="flex" justifyContent="center">
          <Typography color="primary" variant="h4" component="h5" gutterBottom>
            {this.props.title}
          </Typography>
        </Box>
        <Box
          display="flex"
          flexDirection="column"
          justifyContent="center"
          m={1}
          p={1}>
          <Box p={1}>
            <FormControl fullWidth>
              <TextField
                label={language.email}
                color="secondary"
                id="email"
                type="email"
                error={!this.state.validEmail}
                onChange={this.emailChange}
                helperText={this.state.helpEmail}
              />
            </FormControl>
          </Box>
          <Box p={1}>
            <FormControl fullWidth>
              <TextField
                id="password"
                label={language.password}
                color="secondary"
                type="password"
                error={!this.state.validPassword}
                onChange={this.passwordChange}
                helperText={this.state.helpPassword}
              />
            </FormControl>
          </Box>

          <Box p={1} m={2} display="flex" justifyContent="flex-end">
            <Link color="secondary" onClick={this.props.onForgot}>
              <Typography variant="caption">{language.forgot}</Typography>
            </Link>
          </Box>

          <Box p={1} m={1} display="flex" justifyContent="flex-end">
            <Button
              variant="outlined"
              color="primary"
              disabled={!this.isValid()}
              onClick={this.loginClick}>
              {language.login}
            </Button>
          </Box>
        </Box>
      </Box>
    );
  }

  emailChangeHandle(event) {
    if (this.valid.email(event.target.value)) {
      this.setState({
        validEmail: true,
        helpEmail: '',
        email: event.target.value,
      });
    } else {
      this.setState({ validEmail: false, helpEmail: language.invalidEmail });
    }
  }

  passwordChangeHandle(event) {
    if (!this.valid.isEmpty(event.target.value)) {
      this.setState({
        validPassword: true,
        helpPassword: '',
        password: event.target.value,
      });
    } else {
      this.setState({
        validPassword: false,
        helpPassword: language.invalidPassword,
      });
    }
  }

  loginHandle(event) {
    this.props.onLogin({
      email: this.state.email,
      password: this.state.password,
    });
    event.preventDefault();
  }

  isValid() {
    return this.state.validEmail && this.state.validPassword;
  }
}

export default withUtils(styles)(SignIn);
