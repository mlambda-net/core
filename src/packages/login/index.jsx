export Register from './register';
export SignIn from './signin';
export Profile from './profile'
export ChangePassword from './changePassword'
export Recovery from './recovery'
