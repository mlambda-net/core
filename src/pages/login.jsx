import React from 'react';
import Box from '@material-ui/core/Box';
import login from '../assets/img/login.png';
import { LeftImage } from '@mlambda-net/core/common';
import { SignIn } from '@mlambda-net/core/login';
import { withUtils } from '@mlambda-net/core/utils';
import Recovery from '@mlambda-net/core/login/recovery';
import { Action, State } from '@mlambda-net/core/common';

const styles = (themes) => ({
  root: {
    display: 'flex',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    width: '100%',
  },
  login: {},
});

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.route = this.props.route;
    this.state = { actual: 'login' };
    this.onLogin = this.onLoginHandle.bind(this);
    this.onForgot = this.onForgotHandle.bind(this);
  }

  login(data) {
    this.route.to('store');
  }

  render() {
    const { classes } = this.props;

    return (
      <Box className={classes.root}>
        <Box width={'500px'}>
          <LeftImage img={<img src={login} width="100%" alt="img" />}>
            <Action actual={this.state.actual}>
              <State name="login">
                <SignIn
                  onForgot={() => this.setState({ actual: 'recovery' })}
                  onLogin={(user) => this.login(user)}
                />
              </State>
              <State name="recovery">
                <Recovery onClose={() => this.setState({ actual: 'login' })} />
              </State>
            </Action>
          </LeftImage>
        </Box>
      </Box>
    );
  }

  onLoginHandle(data) {
    console.log(data);
  }

  onForgotHandle() {}
}

export default withUtils(styles)(Login);
