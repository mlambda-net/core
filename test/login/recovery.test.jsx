import React from 'react';
import renderer from 'react-test-renderer';
import { expect } from '@jest/globals';
import { Recovery } from '@mlambda-net/core/login';

test('Recovery render', () => {
  const item = renderer.create(<Recovery />);
  let tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});
