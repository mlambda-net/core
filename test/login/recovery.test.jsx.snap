// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recovery render 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-8atqhb{width:100%;}",
      }
    }
    data-emotion="css 8atqhb"
  />,
  <div
    className="MuiBox-root css-8atqhb"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1l4w6pd{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}",
        }
      }
      data-emotion="css 1l4w6pd"
    />
    <div
      className="MuiBox-root css-1l4w6pd"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1htymla-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:2.125rem;line-height:1.235;letter-spacing:0.00735em;margin-bottom:0.35em;color:#1976d2;}",
          }
        }
        data-emotion="css 1htymla-MuiTypography-root"
      />
      <h5
        className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1htymla-MuiTypography-root"
      >
        Recovery
      </h5>
    </div>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-hpgf8j{padding:8px;}",
        }
      }
      data-emotion="css hpgf8j"
    />
    <div
      className="MuiBox-root css-hpgf8j"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-q8hpuo-MuiFormControl-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;position:relative;min-width:0;padding:0;margin:0;border:0;vertical-align:top;width:100%;}",
          }
        }
        data-emotion="css q8hpuo-MuiFormControl-root"
      />
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1u3bzj6-MuiFormControl-root-MuiTextField-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;position:relative;min-width:0;padding:0;margin:0;border:0;vertical-align:top;}",
            }
          }
          data-emotion="css 1u3bzj6-MuiFormControl-root-MuiTextField-root"
        />
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1x92c1k-MuiInputLabel-root{display:block;transform-origin:top left;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;max-width:calc(100% - 24px);position:absolute;left:0;top:0;-webkit-transform:translate(14px, 16px) scale(1);-moz-transform:translate(14px, 16px) scale(1);-ms-transform:translate(14px, 16px) scale(1);transform:translate(14px, 16px) scale(1);-webkit-transition:color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;transition:color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;z-index:1;pointer-events:none;}",
              }
            }
            data-emotion="css 1x92c1k-MuiInputLabel-root"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root{color:rgba(0, 0, 0, 0.6);font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.4375em;letter-spacing:0.00938em;padding:0;position:relative;display:block;transform-origin:top left;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;max-width:calc(100% - 24px);position:absolute;left:0;top:0;-webkit-transform:translate(14px, 16px) scale(1);-moz-transform:translate(14px, 16px) scale(1);-ms-transform:translate(14px, 16px) scale(1);transform:translate(14px, 16px) scale(1);-webkit-transition:color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;transition:color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;z-index:1;pointer-events:none;}.css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root.Mui-focused{color:#9c27b0;}.css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root.Mui-disabled{color:rgba(0, 0, 0, 0.38);}.css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root.Mui-error{color:#d32f2f;}",
              }
            }
            data-emotion="css 1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
          />
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary Mui-error MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            email
          </label>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": "@-webkit-keyframes mui-auto-fill{from{display:block;}}@keyframes mui-auto-fill{from{display:block;}}@-webkit-keyframes mui-auto-fill-cancel{from{display:block;}}@keyframes mui-auto-fill-cancel{from{display:block;}}",
              }
            }
            data-emotion="css-global 1prfaxn"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.4375em;letter-spacing:0.00938em;color:rgba(0, 0, 0, 0.87);box-sizing:border-box;position:relative;cursor:text;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;border-radius:4px;}.css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root.Mui-disabled{color:rgba(0, 0, 0, 0.38);cursor:default;}.css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.87);}@media (hover: none){.css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.23);}}.css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline{border-color:#9c27b0;border-width:2px;}.css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root.Mui-error .MuiOutlinedInput-notchedOutline{border-color:#d32f2f;}.css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root.Mui-disabled .MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.26);}",
              }
            }
            data-emotion="css hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
          />
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary Mui-error MuiInputBase-formControl css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input{font:inherit;letter-spacing:inherit;color:currentColor;padding:4px 0 5px;border:0;box-sizing:content-box;background:none;height:1.4375em;margin:0;-webkit-tap-highlight-color:transparent;display:block;min-width:0;width:100%;-webkit-animation-name:mui-auto-fill-cancel;animation-name:mui-auto-fill-cancel;-webkit-animation-duration:10ms;animation-duration:10ms;padding:16.5px 14px;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input::-webkit-input-placeholder{color:currentColor;opacity:0.42;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input::-moz-placeholder{color:currentColor;opacity:0.42;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:-ms-input-placeholder{color:currentColor;opacity:0.42;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input::-ms-input-placeholder{color:currentColor;opacity:0.42;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:focus{outline:0;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:invalid{box-shadow:none;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input::-webkit-search-decoration{-webkit-appearance:none;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input::-webkit-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input::-moz-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:-ms-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input::-ms-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:focus::-webkit-input-placeholder{opacity:0.42;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:focus::-moz-placeholder{opacity:0.42;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:focus:-ms-input-placeholder{opacity:0.42;}label[data-shrink=false]+.MuiInputBase-formControl .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:focus::-ms-input-placeholder{opacity:0.42;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input.Mui-disabled{opacity:1;-webkit-text-fill-color:rgba(0, 0, 0, 0.38);}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:-webkit-autofill{-webkit-animation-duration:5000s;animation-duration:5000s;-webkit-animation-name:mui-auto-fill;animation-name:mui-auto-fill;}.css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input:-webkit-autofill{border-radius:inherit;}",
                }
              }
              data-emotion="css 1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            />
            <input
              aria-invalid={true}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="email"
            />
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-rki20w-MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.23);}",
                }
              }
              data-emotion="css rki20w-MuiOutlinedInput-notchedOutline"
            />
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1d3z3hw-MuiOutlinedInput-notchedOutline{text-align:left;position:absolute;bottom:0;right:0;top:-5px;left:0;margin:0;padding:0 8px;pointer-events:none;border-radius:inherit;border-style:solid;border-width:1px;overflow:hidden;min-width:0%;border-color:rgba(0, 0, 0, 0.23);}",
                }
              }
              data-emotion="css 1d3z3hw-MuiOutlinedInput-notchedOutline"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-yjsfm1{float:unset;width:auto;overflow:hidden;display:block;padding:0;height:11px;font-size:0.75em;visibility:hidden;max-width:0.01px;-webkit-transition:max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;transition:max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;white-space:nowrap;}.css-yjsfm1>span{padding-left:5px;padding-right:5px;display:inline-block;opacity:0;visibility:visible;}",
                  }
                }
                data-emotion="css yjsfm1"
              />
              <legend
                className="css-yjsfm1"
              >
                <span>
                  email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-170iay{padding:8px;margin:8px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-ms-flex-pack:end;-webkit-justify-content:flex-end;justify-content:flex-end;}",
        }
      }
      data-emotion="css 170iay"
    />
    <div
      className="MuiBox-root css-170iay"
    >
      <div
        className="MuiBox-root css-hpgf8j"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-10tjo1u-MuiButton-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(25, 118, 210, 0.5);color:#1976d2;}.css-10tjo1u-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);border:1px solid #1976d2;}@media (hover: none){.css-10tjo1u-MuiButton-root:hover{background-color:transparent;}}.css-10tjo1u-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
            }
          }
          data-emotion="css 10tjo1u-MuiButton-root"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1rwt2y5-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(25, 118, 210, 0.5);color:#1976d2;}.css-1rwt2y5-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-1rwt2y5-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-1rwt2y5-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-1rwt2y5-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);border:1px solid #1976d2;}@media (hover: none){.css-1rwt2y5-MuiButtonBase-root-MuiButton-root:hover{background-color:transparent;}}.css-1rwt2y5-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
            }
          }
          data-emotion="css 1rwt2y5-MuiButtonBase-root-MuiButton-root"
        />
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          cancel
        </button>
      </div>
      <div
        className="MuiBox-root css-hpgf8j"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          Recovery
        </button>
      </div>
    </div>
  </div>,
]
`;
